src/rgw/rgw_process.cc

```diff
@@ -305,6 +306,11 @@ int process_request(rgw::sal::RGWRadosStore* const store,
     *http_ret = s->err.http_ret;
   }
   int op_ret = 0;
+
+  if (user && rgw::sal::RGWObject::empty(s->user)) {
```

Comment:
Should be !rgw::sal::

Reply:
@dang @cbodley `rgw::sal::RGWObject::empty()` won't take a `s->user` (https://jenkins.ceph.com/job/ceph-pull-requests/63777/consoleFull#779856932a811ea2-3e7b-466b-84b4-d13df7e35809) would `if (user && s->user)` work?

Reply:
@cbodley if you just run a `curl -L` on the rgw the user that shows up is `anonymous`. Does this imply that `s->user` gets initialized for all requests?

Reply:
The proper check is:

if (user && !rgw::sal::RGWObject::empty(s->user.get())) {